implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala

```diff
@@ -140,7 +140,9 @@ object Macros {
           if (isEnum) {
             // brute force to get the outer class -- @todo maybe there's a better way...
             val getOuter = c.parse(argType.toString.replace(".Value", ""))
-            q"implicitly[${c.prefix}.To[String]].map($getOuter.withName(_))"
+            q"""val nameToValue = scala.collection.mutable.Map.empty[String, $argType] ++ $getOuter.values.map(v => v.toString -> v)
+                implicitly[${c.prefix}.To[String]].map(nameToValue)
+             """
```

Comment:
Please consider to use `CHM` instead to lookup properly in case of dynamic extension of the enum set:
```
scala> object LocationType extends Enumeration { val GPS = Value(1, "GPS"); val IP = Value(2, "IP"); def more(name: String) = Value(nextId, name) }
defined object LocationType

scala> LocationType.values
res0: LocationType.ValueSet = LocationType.ValueSet(GPS, IP)

scala> LocationType.more("Postal Address")
res1: LocationType.Value = Postal Address

scala> LocationType.values
res2: LocationType.ValueSet = LocationType.ValueSet(GPS, IP, Postal Address)

scala> val enumCache = new java.util.concurrent.ConcurrentHashMap[String, LocationType.Value]()
enumCache: java.util.concurrent.ConcurrentHashMap[String,LocationType.Value] = {}

scala> def withName(name: String): LocationType.Value = enumCache.computeIfAbsent(name, LocationType.withName)
withName: (name: String)LocationType.Value

scala> withName("GPS")
res3: LocationType.Value = GPS

scala> LocationType.more("Address")
res4: LocationType.Value = Address

scala> withName("Address")
res5: LocationType.Value = Address

scala> withName("XXX")
java.util.NoSuchElementException: No value found for 'XXX'
  at scala.Enumeration.$anonfun$withName$2(Enumeration.scala:159)
  at scala.Enumeration.withName(Enumeration.scala:159)
  at .$anonfun$withName$1(<console>:1)
  at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
  at .withName(<console>:1)
  ... 28 elided
```

Reply:
Moreover, `java.util.concurrent.ConcurrentHashMap` stores hash codes separately so will return cached values faster than scala.collection.mutable.HashMap. Yet another `CHM`'s feature that the table for backets will be created on the first call of computeIfAbsent, not upfront.

Reply:
Oh, I like that, thanks!

Reply:
But defining an enumeration like that is a bit pathological... not really an enumeration anymore. If we assume a static set of values, it seems like pre-loading the values would be more performant (in the short run -- no diff amortized in the long run). I do like the idea that enumerations you never use don't take up extra memory.

Reply:
If you never use them, I don't think the classloader would even load them.

Reply:
I meant "never use" in terms of the values encountered in the source data being deserialized (like Json), not code references.

Reply:
```scala
import com.rallyhealth.weejson.v1.jackson.FromJson
import com.rallyhealth.weepickle.v1.WeePickle
import com.rallyhealth.weepickle.v1.WeePickle._

object Suit extends Enumeration {

  val Spades = Value

  println("Suit initialized.")

  implicit val pickler = fromTo[String].bimap[Value](_.toString, withName)
}

case class Card(name: String, suit: Option[Suit.Value] = None)
object Card {
  implicit val pickler = WeePickle.macroFromTo[Card]
}

object Test extends App {

  val card = FromJson("""{"name": "Joker"}""").transform(ToScala[Card])

  println("Card with no Suit: " + card)
  println("-----------------------")
  println("Forcing classloading of Suit...")

  val forceClassloading = Suit.Spades
}
```

```
Card with no Suit: Card(Joker,None)
-----------------------
Forcing classloading of Suit...
Suit initialized.
```